include/launch_core/os/fs.h

```diff
@@ -67,13 +67,11 @@ int32_t l_fs_close(l_fs_file_t *p_file);
  * @brief Seek to the appropriate (l_fs_file_t *p_filebased on the logging mode) point and write
  * the buffer
  *
- * @param src 	buffer to write (l_fs_file_t *p_fileassumed fixed-width, with the initialized width)
- *
- * @retval = 0 			success
- * @retval = -ENOTINIT 	logger was never initialized
- * @retval < 0			other errno from fs functions
+ * @param[in] p_file 	pointer to file handle
+ * @param[in] src 	buffer to write (l_fs_file_t *p_fileassumed fixed-width, with the initialized width)
+ * @param[out] err_flag 	Pointer to error flag
```

Comment:
Is retval number of bytes written or something else? Probably worth a doxygen comment. 
Also, it might be worth specifying the logger specific ENOINIT in the err_flag comment. 

Update after reading fs.c:
Is there any reason to use an error pointer sometimes for some errors and return code other times? It looks like both pointer and return can give errors so why not just consolidate both into one error. 

If we want to discriminate between logger error path vs. file system error path probably want a comment for that .

Reply:
size_t generally means bytes read/written. Having a pointer to an error is also common. Usually things where you read and write you want to return bytes written/read. If you need to also report an error, then have a pointer to it. If you don't need to return bytes written/read then you can return an error code. A write/read of 0 can signify that an issue might've occurred so the user should check the error status.

Reply:
In any case, the return doxy is added